packages/frontend/src/ee/features/customRoles/CustomRolesEditModal.tsx

```diff
@@ -0,0 +1,118 @@
+import {
+    Button,
+    Group,
+    Modal,
+    Stack,
+    TextInput,
+    Textarea,
+} from '@mantine/core';
+import { useForm } from '@mantine/form';
+import { type FC, useEffect } from 'react';
+
+import { type RoleWithScopes } from '@lightdash/common';
+
+type EditModalProps = {
+    isOpen: boolean;
+    onClose: () => void;
+    onSave: (values: { name: string; description?: string }) => void;
+    isWorking?: boolean;
+    role: RoleWithScopes;
+};
+
+export const CustomRolesEditModal: FC<EditModalProps> = ({
```

Comment:
this component can probably be combined with CreateModal

Reply:
In the next PR, I have two separate pages for Create and Edit that are a bit more different in how they create handlers, but return the same component.